File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.py
class Solution:
    def minimumAbsDifference(self, arr: List[int]) -> List[List[int]]:
        res = {}
        arr.sort()
        for i in  range(len(arr)-1) :
                j = i+1
                x  = arr[j] -  arr[i]
                if x in res :
                    res[x].append([arr[i],arr[j]])
                else :
                    res[x] = [[arr[i],arr[j]]]
        #print(res)
        res = dict(sorted(res.items() , key = lambda x:x[0]))
        #print(res)
        return list(res.values())[0]

File: 1200-minimum-absolute-difference/1200-minimum-absolute-difference.cpp
class Solution {
public:
      bool cmp(pair<int,vector<vector<int>>>& a,
         pair<int,vector<vector<int>>>& b)
{
    return a.first < b.first;
}
    vector<vector<int>> minimumAbsDifference(vector<int>& arr) {
    vector<vector<int>>  ans;
    sort(arr.begin(),arr.end());
    int min = INT_MAX;
      for (int i =0 ;i < arr.size()-1;i++){
          int j =  i+1;
          int x = arr[j] - arr[i];
            if(x<min)
            {
            min = x;
            ans.clear();
            ans.push_back({arr[i],arr[j]});
            }
            else if(arr[j]-arr[i]==min)
                ans.push_back({arr[i],arr[j]});
          
      }
        return ans;
    }
  
};